category/category.go

```diff
@@ -0,0 +1,255 @@
+	db = m.db.Model(&relationship).Where("category_id=?", categoryID).Find(&relationship)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"categories_id": categoryID,
+		}, "workitemtypes of category not found")
+		return nil, errors.NewNotFoundError("work item type category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	return relationship, nil
+}
+
+// LoadCategoryFromDB returns category for the given id
+// This is needed to check if a category is present in db or not.
+func (m *GormRepository) LoadCategoryFromDB(ctx context.Context, id uuid.UUID) (*Category, error) {
+	log.Logger().Infoln("Loading category", id)
+	res := Category{}
+	db := m.db.Model(&res).Where("id=?", id).First(&res)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": id,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", id.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"categoryID": id,
+		}, "category retrieval error", err.Error())
+		return nil, errors.NewInternalError(err.Error())
+	}
+	return &res, nil
+}
+
+// LoadWorkItemTypeCategoryRelationship loads all the relationships of a category. This is required for testing.
+func (m *GormRepository) LoadWorkItemTypeCategoryRelationship(ctx context.Context, workitemtypeID uuid.UUID, categoryID uuid.UUID) (*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	getCategory := Category{}
+	db := m.db.Model(&getCategory).Where("id=?", categoryID).Find(&getCategory)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"categories_id": categoryID,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	relationship := WorkItemTypeCategoryRelationship{}
+	db = m.db.Model(&relationship).Where("category_id=? AND workitemtype_id=?", categoryID, workitemtypeID).Find(&relationship)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"categories_id": categoryID,
+		}, "workitemtypes of category not found")
+		return nil, errors.NewNotFoundError("work item type category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	return &relationship, nil
+}
+
+// Save saves a category object. This function is used to update category properly during migration -> createOrUpdateSingleCategory()
+func (m *GormRepository) Save(ctx context.Context, category *Category) (*Category, error) {
+	res := Category{}
+	log.Info(ctx, map[string]interface{}{
+		"id": category.ID,
+	}, "Looking for category")
+	tx := m.db.Model(&res).Where("id=?", category.ID).First(&res)
+	if tx.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"id": category.ID,
```

Comment:
in order to easily trace the logs, you should use `category_id` instead of `id` in these logs. Please change it accordingly in all the document